c7n/resources/sns.py

```diff
@@ -46,6 +47,35 @@ class resource_type(object):
         )
 
 
+class SNSPolicyChecker(PolicyChecker):
+
+    @property
+    def allowed_endpoints(self):
+        return self.checker_config.get('allowed_endpoints', ())
+
+    @property
+    def allowed_protocols(self):
+        return self.checker_config.get('allowed_protocols', ())
+
+    def handle_sns_endpoint(self, s, c):
+        conditions = self.normalize_conditions(s)
+        # yield to aws:sourceowner
+        if not self.allowed_endpoints:
+            return not any(
+                single_condition.get('key',None) == 'aws:sourceowner'
```

Comment:
this seems less then clear or to semantic intent when specifying endpoints, allowed_endpoints is being treated as a boolean here, sourceowner isn't being checked on its value, whats the intent if a user doesn't specify endpoints?

Reply:
The thought process is that if `allowed_endpoints` is not explicitly whitelisted then we check that `aws:sourceowner` is a present key so that, implicitly, the validity of the policy is handled by the `handle_aws_sourceowner()` method.

Reply:
ic, sounds good